apps/src/client.rs

```diff
@@ -227,50 +251,57 @@ pub fn connect(
                 break 'read;
             }
 
-            let (len, from) = match socket.recv_from(&mut buf) {
-                Ok(v) => v,
-
-                Err(e) => {
-                    // There are no more UDP packets to read, so end the read
-                    // loop.
-                    if e.kind() == std::io::ErrorKind::WouldBlock {
-                        trace!("recv() would block");
-                        break 'read;
-                    }
-
-                    return Err(ClientError::Other(format!(
-                        "recv() failed: {:?}",
-                        e
-                    )));
-                },
-            };
+            for event in &events {
+                let token: usize = event.token().into();
+                let socket = &sockets[token];
+                let (len, from) = match socket.recv_from(&mut buf) {
+                    Ok(v) => v,
+
+                    Err(e) => {
+                        // There are no more UDP packets to read, so end the read
+                        // loop.
```

Comment:
Is this comment true? Isn't there a possibility that there are other packets to read from other sockets?

Reply:
Good catch! Indeed, we should instead iterate over the events and loop on a given socket until there is nothing more to read on that one. Will update the code.